spec/services/processors/external/contributions_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+RSpec.describe Processors::External::Contributions do
```

Comment:
it could seems to be useless but since it is the "entry point" of the worker it is just the handler/caller of the other processors for every username we have in DB

Reply:
Yeah, I understand why you did it that way. But we are then checking _how_ and not _what_.
I think we could get rid of the mocks easily.
If we are checking that `Builders::ExternalProject` is called, we could assert that an external project for the user is created. And as for `Processors::External::PullRequests`, we could assert external pull requests have been created.
We don't need to go through all the scenarios those classes handle (like if those records existed before), we just need to make sure those records exist for all the users in the DB.
What do you think?